File: atcoder.jp/contests/abc135/tasks/abc135_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

input();
const P = input().split(" ").map(Number);

const cnt = P.filter((p, i) => i + 1 !== p).length;
const ans = cnt === 0 || cnt === 2;

console.log(ans ? "YES" : "NO");

File: atcoder.jp/contests/abc135/tasks/abc135_b/main.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        p: [usize; n]
    }

    let cnt = p
        .into_iter()
        .enumerate()
        .filter(|(i, p)| i + 1 != *p)
        .count();

    let ans = cnt == 0 || cnt == 2;

    println!("{}", if ans { "YES" } else { "NO" });
}
